proposals/params-collections.md

```diff
@@ -0,0 +1,171 @@
+  - A type parameter is less specific than a non-type parameter.
+  - Recursively, a constructed type is more specific than another constructed type (with the same number of type arguments) if at least one type argument is more specific and no type argument is less specific than the corresponding type argument in the other.
+  - An array type is more specific than another array type (with the same number of dimensions) if the element type of the first is more specific than the element type of the second.
+- Otherwise if one member is a non-lifted operator and the other is a lifted operator, the non-lifted one is better.
+- If neither function member was found to be better, and all parameters of `Mᵥ` have a corresponding argument whereas default arguments need to be substituted for at least one optional parameter in `Mₓ`, then `Mᵥ` is better than `Mₓ`.
+- If for at least one parameter `Mᵥ` uses the ***better parameter-passing choice*** ([§12.6.4.4](expressions.md#12644-better-parameter-passing-mode)) than the corresponding parameter in `Mₓ` and none of the parameters in `Mₓ` use the better parameter-passing choice than `Mᵥ`, `Mᵥ` is better than `Mₓ`.
+- Otherwise, no function member is better.
+
+### Ref safety
+
+The [collection expressions ref safety section](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#ref-safety) is applicable to
+the construction of parameter collections when APIs are invoked in their expanded form.
+
+## Open questions
+
+### Stack allocations 
+
+Here is a quote from https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#unresolved-questions:
+"Stack allocations for huge collections might blow the stack.  Should the compiler have a heuristic for placing this data on the heap?
+Should the language be unspecified to allow for this flexibility?
+We should follow the spec for [`params Span<T>`](https://github.com/dotnet/csharplang/issues/1757)." It sounds like we have to answer
+the questions in context of this proposal.
+
+## Alternatives 
+
+There is an alternative [proposal](https://github.com/dotnet/csharplang/blob/main/proposals/params-span.md) that extends
+`params` only for `ReadOnlySpan<T>`.
+
+Also, one might say, that with [collection expressions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md)
+now in the language, there is no need to extend `params` support at all. Foe any collection type. To consume an API with collection type, a developer
+simply needs to add two characters, `[` before the expanded list of arguments, and `]` after it. Given that, extending `params` support might be an overkill,
+especially that other languages are unlikely to support consumption of non-array `params` parameters any time soon.
```

Comment:
right.  this was the POV of a few people the last time we discussed this.  namely, that it's totally fine to just have the user write `[...]` around teh arguments.  THe critical case as `params Span` because in that case we want the API developer to be able to add the overload and have it take precedence over the array version, just by recompiling (so no change to the callsite). 

--

Note: i'm personally fine going either way.  `params collection` seems the most consistent approach to take.  but `params span` also seems sufficient.

Reply:
It's probably worth expanding the "we at least need to do `params Span` so the BCL can offer better overloads" part explicitly here.

Reply:
The better overloads part is definitely a big motivator for this feature.